app/queues/common/expiration-request.server.ts

```diff
@@ -37,23 +42,41 @@ const expirationRequestWorker = new Worker(
       logger.info('process DNS record expiration');
       let dnsRecords = await getExpiredDnsRecords();
       await Promise.all(
-        dnsRecords.map(async ({ id, user }) =>
+        dnsRecords.map(async ({ id, subdomain, user }) =>
           Promise.all([
-            // delete records from Route53 and DB
+            // delete DNS records from DB
             deleteDnsRecordById(id),
-            // add notification jobs (assuming deletion went successfully)
+            // add notification jobs
             addNotification({
               emailAddress: user.email,
-              subject: 'DNS record expiration subject',
-              message: 'DNS record expiration message',
+              subject: 'My.Custom.Domain DNS record expired',
+              message: `${user.displayName}, your DNS record with subdomain: ${subdomain} has expired.`,
             }),
           ])
         )
       );
     } catch (err) {
       throw new UnrecoverableError(`Unable to process DNS record expiration: ${err}`);
     }
-    logger.info('TODO: process certificate expiration');
+    try {
+      logger.info('process certificate expiration');
+      let certificates = await getExpiredCertificates();
+      certificates.map(async ({ id, username, domain, user }) => {
+        let mostRecentCertificate = await getCertificateByUsername(username);
+        // add notification jobs
+        if (mostRecentCertificate.id === id) {
+          await addNotification({
+            emailAddress: user.email,
+            subject: 'My.Custom.Domain certificate expired',
+            message: `${user.displayName}, your certificate with domain: ${domain} has expired.`,
+          });
+          // delete certificate from DB
+          await deleteCertificateById(id);
```

Comment:
You should still delete it, even if it was not the most recent. If the cert is expired, there is no point in keeping it, it is useless, cannot have any reason to revoke it, etc.

Reply:
That's by accident 😛

Reply:
:P